enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,210 @@
+* Upgrade blocking mechanism must be generic such that it can be repurposed in the future.
+* It must be clear what an administrator is acknowledging.
+* Administrators can bypass the upgrade blocking mechanism by using the existing CVO "force" implementation.
+* The UI will clearly indicate when this upgrade blocking mechanism is blocking upgrades, why, and what needs to be done to clear the block.
+* This upgrade blocking mechanism must be compatible with SD, ACM, IKS/ROKS, Hive, and Hypershift.
+
+### Non-Goals
+
+## Proposal
+
+### General Implementation
+
+CVO's existing framework for blocking minor level upgrades will be leveraged.
+The existing framework periodically checks upgradeable conditions.
+A new [upgradeable check method][upgradeableCheck-interface] will be added to the existing framework to implement this upgrade blocking mechanism.
+Use of this existing framework also allows this new upgrade blocking mechanism to be bypassed by using the `force` upgrade option.
+
+The new method's logic will be driven by configmap keys each of which represents a potential minor level upgrade blocker.
+CVO will consume a new configmap, `admin-gates`, defining the keys. For example:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: admin-gates
+  namespace: openshift-cluster-version
+data:
+  ack-4.8-kube-122-api-removals-in-4.9: |
+     Detailed description of what admin needs to do...Read documentation here https://
+  ack-4.8-gate-2: |
+     Detailed description of what admin needs to do...
+```
+The key name will include an OCP major/minor version number that indicates to which OCP version the key applies.
+Key names must be in the format `ack-X.Y-<description>` where `X.Y` is the major/minor version number and `<description>` is words describing to what the administrator is agreeing.
+For example, a key name beginning with `ack-4.8` would only apply to an OCP cluster currently at version 4.8.
+The key's value will be a detailed message pointing to relevant OCP documentation explaining what the administrator is expected to do before acknowledging task completion.
+If running within the applicable version, CVO will check a newly defined configmap, `admin-acks`, for the key.
+The key represents a specific upgrade blocking condition that must be acknowledged before a minor level upgrade is allowed.
+An administrator acknowledges a condition by adding the corresponding key with a value of `true` to the configmap `admin-acks`.
+Values other than `true` will be ignored and therefore not clear the upgrade block.
+The keys will be named to make it explicitly clear what the administrator is acknowledging and thereby agreeing to.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to the next minor release until it finds each of the applicable predefined keys in the configmap with a value of `true`.
```

Comment:
Just to clarify, is this `Upgradeable=false` condition going to be set on the ClusterVersion status? Currently the Upgradable condition seems to only be set on ClusterOperator status(es)

Reply:
i'll be honest, i thought the CVO already had its own clusteroperator resource but I see that it does not seem to.  We could certainly introduce one, but since @sdodson and @jottofar got this far w/o raising the question i'm going to hope that means they had something else in mind.

Reply:
CVO currently sets ClusterOperatorStatusCondition `Upgradeable=false` when cluster version overrides exist or there is a pending CVO initiated resource deletion (via the recently added manifest deletion support). This would add an additional one for ack(s).

Reply:
sets it where?  on ClusterVersion?

Reply:
On the cluster-version operator's status, correct @jottofar ?

Reply:
yup, we have `Upgradeable` on the ClusterVersion, and it's a union of what we collect from ClusterOperator and the CVO's own opinions on things like this (currently the presence of meaningful `overrides` in ClusterVersion's `spec`.  And any in-progress deletions.  [Source][1]).

[1]: https://github.com/openshift/cluster-version-operator/blob/d58378108819cf8455600521bc5186acb7d201da/pkg/cvo/upgradeable.go#L239-L241

Reply:
@wking if the Upgradeable condition is not present, can we assume that it is true?

Reply:
If I understand your question correctly @csrwng, the condition we're proposing to add in this enhancement, which would be added [here](https://github.com/openshift/cluster-version-operator/blob/d58378108819cf8455600521bc5186acb7d201da/pkg/cvo/upgradeable.go#L238), can be nil which is checked [here](https://github.com/openshift/cluster-version-operator/blob/d58378108819cf8455600521bc5186acb7d201da/pkg/cvo/upgradeable.go#L37).

Reply:
Thanks, from a hypershift perspective I just want to make sure that we have a clear signal as to whether the cluster is upgradeable by either looking at ClusterVersion or ClusterOperator(s). We don't rely on the CVO to perform an upgrade, but we want to report in our status whether the cluster can be upgraded and let the service provider make the decision whether to force it or not.